Extension/src/githubAPI.ts

```diff
@@ -119,33 +164,34 @@ function getTargetBuild(updateChannel: string, builds: Build[]): Build {
     return needsUpdate(userVersion, targetVersion) ? targetBuild : undefined;
 }
 
+/**
+ * Download and parse the release list JSON from the GitHub API into a Build[].
+ * @return Information about the released builds of the C/C++ extension.
+ */
 async function getReleaseJson(): Promise<Build[]> {
     return new Promise<Build[]>((resolve, reject) => {
-        // Create temp file to hold json
+        // Create temp file to hold JSON
         tmp.file((err, releaseJsonPath, fd, cleanupCallback) => {
             if (err) {
                 return reject(new Error('Failed to create release json file'));
             }
 
-            const releaseUrl: string = 'https://api.github.com/repos/Microsoft/vscode-cpptools/releases';
-            const header: OutgoingHttpHeaders = { 'User-Agent': 'vscode-cpptools' };
-            const downloadReleaseJson: any = util.downloadFileToDestination.bind(undefined, releaseUrl, releaseJsonPath, header);
-            const rejectDownload: any = () => { return reject(new Error('Failed to download release json')); };
-
-            const readJson: any = util.readFileText.bind(undefined, releaseJsonPath);
-            const rejectRead: any = () => { return reject(new Error('Failed to read release json file')); };
-
-            const rejectParse: any = () => { return reject(new Error('Failed to parse release json')); };
+            // Helper functions to handle promise rejection
+            const rejectDownload: any = () => { return reject(new Error('Failed to download release JSON')); };
```

Comment:
Avoid using `any` when possible.  This is masking a bug in the `.then` chain below.